tests/scripts/generate_code.py

```diff
@@ -0,0 +1,710 @@
+"""
+  Test suites code generator.
+
+  Copyright (C) 2006-2017, ARM Limited, All Rights Reserved
+  SPDX-License-Identifier: Apache-2.0
+
+  Licensed under the Apache License, Version 2.0 (the "License"); you may
+  not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  This file is part of mbed TLS (https://tls.mbed.org)
+"""
+
+import os
+import re
+import argparse
+import shutil
+
+
+"""
+Generates code in following structure.
+
+<output dir>/
+ |-- mbedtls/
+ |   |-- <test suite #1>/
+ |   |    |-- main.c
+ |   |    |-- *.data files
+ |   ...
+ |   |-- <test suite #n>/
+ |   |    |-- main.c
+ |   |    |-- *.data files
+ |   |
+"""
+
+
+BEGIN_HEADER_REGEX = '/\*\s*BEGIN_HEADER\s*\*/'
+END_HEADER_REGEX = '/\*\s*END_HEADER\s*\*/'
+
+BEGIN_DEP_REGEX = 'BEGIN_DEPENDENCIES'
+END_DEP_REGEX = 'END_DEPENDENCIES'
+
+BEGIN_CASE_REGEX = '/\*\s*BEGIN_CASE\s*(.*?)\s*\*/'
+END_CASE_REGEX = '/\*\s*END_CASE\s*\*/'
+
+
+class InvalidFileFormat(Exception):
+    """
+    Exception to indicate invalid file format. 
+    """
+    pass
+
+
+class FileWrapper(file):
```

Comment:
ignore my previous comment. yes it can be used like this:

```
def three_to_five(e):
    for i, l in e:
        print "three_to_five (%d) %s" % (i, l)
        if l == "five\n":
            break


with open("123.txt", "r") as f:
    e = enumerate(f)
    for i, l in e:
        print "(%d) %s" % (i, l)
        if l == "three\n":
            three_to_five(e)
```